### PR TITLE
add image zoom component

### PR DESCRIPTION
**Pull request by jellli**
## Add build-in component to zoom image
I'm not sure whether it is necessary (and how) to give users more configurability.

Closes: https://github.com/shuding/nextra/issues/2238

## What's being changed (if available, include any code snippets, screenshots, or gifs):
CleanShot 2024-12-19 at 17 39 06
CleanShot 2024-12-19 at 17 37 15

## Check off the following:

- [x] I have reviewed my changes in staging, available via the **View
      deployment** link in this PR's timeline (this link will be available after
      opening the PR).

### CODE REVIEW

**Comment by jellli** [comment]
@dimaMachina need some help with `patches` property of pnpm, I can't install deps because of the same reason with [this](https://github.com/shuding/nextra/actions/runs/12410097629/job/34645013265?pr=3829) unless i delete this line of package.json
```diff
  "pnpm": {
    "overrides": {
      "esbuild": "^0.24.0",
      "next": "15.1.0"
    },
    "patchedDependencies": {
--    "@changesets/assemble-release-plan@6.0.4": "patches/@changesets__assemble-release-plan.patch",
      "eslint-plugin-deprecation@3.0.0": "patches/eslint-plugin-deprecation.patch",
      "tsup@8.3.5": "patches/tsup@8.3.5.patch"
    }
  }
```

**Review by dimaMachina — APPROVED** [review]
thank you for your nice contribution! I will review

File: packages/nextra/styles/rmiz.css [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+  box-shadow: 0 0 1px rgba(255, 255, 255, 0.5);
+  color: #fff;
+  height: 40px;
+  margin: 0;
+  outline-offset: 2px;
+  padding: 9px;
+  touch-action: manipulation;
+  width: 40px;
+  -webkit-appearance: none;
+  -moz-appearance: none;
```
Comment (dimaMachina) [inline packages/nextra/styles/rmiz.css:20]:
```suggestion
```
will be added by lightningcss

File: packages/nextra/styles/rmiz.css [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+}
+
+[data-rmiz-modal-overlay] {
+  position: absolute;
+  inset: 0;
+  transition: background-color 0.3s;
+}
+
+[data-rmiz-modal-overlay='hidden'] {
+  background-color: rgba(255, 255, 255, 0);
```
Comment (dimaMachina) [inline packages/nextra/styles/rmiz.css:81]:
```suggestion
  background-color: #fff;
```

File: packages/nextra/src/client/components/image-zoom.tsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
+'use client'
+
+import Image, { type ImageProps } from 'next/image'
+import React, { type ImgHTMLAttributes } from 'react'
+import Zoom from 'react-medium-image-zoom'
+
+interface ImageZoomProps extends ImageProps {
+  children?: React.ReactNode
```
Comment (dimaMachina) [inline packages/nextra/src/client/components/image-zoom.tsx:8]: `` doesn't have `children`, why we need it?

File: packages/nextra-theme-docs/css/styles.css [hunk 4]
```diff
@@ -8,6 +8,7 @@
 @import 'nextra/styles/cards.css';
 @import './hamburger.css';
 @import './typesetting-article.css';
+@import 'nextra/styles/rmiz.css';
```
Comment (dimaMachina) [inline packages/nextra-theme-docs/css/styles.css:11]:
```suggestion
@import 'nextra/styles/react-medium-image-zoom.css';
```
let's be more precise

File: packages/nextra/styles/react-medium-image-zoom.css [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+  transform-origin: top left;
+  transition: transform 0.3s;
+}
+
+@media (prefers-reduced-motion: reduce) {
+  [data-rmiz-modal-overlay],
+  [data-rmiz-modal-img] {
+    transition-duration: 0.01ms !important;
+  }
+}
```
Comment (dimaMachina) [inline packages/nextra/styles/react-medium-image-zoom.css:107]: maybe just set it to `none`? as we do for twoslash styles? https://github.com/shuding/nextra/blob/ac70dfedfec29f3634a333a282ee94c1e0765cb5/packages/nextra/styles/code-block.css#L89-L94

**Review by dimaMachina — CHANGES_REQUESTED** [review]
https://github.com/user-attachments/assets/99a5a1b7-eb33-4d64-930b-9eb25d0565ad

need to fix images statically imported in cards

https://github.com/shuding/nextra/blob/ac70dfedfec29f3634a333a282ee94c1e0765cb5/docs/pages/showcase.mdx?plain=1#L12-L14

**Comment by dimaMachina** [comment]
> @dimaMachina need some help with `patches` property of pnpm, I can't install deps because of the same reason with [this](https://github.com/shuding/nextra/actions/runs/12410097629/job/34645013265?pr=3829) unless i delete this line of package.json
> 
> ```diff
>   "pnpm": {
>     "overrides": {
>       "esbuild": "^0.24.0",
>       "next": "15.1.0"
>     },
>     "patchedDependencies": {
> --    "@changesets/assemble-release-plan@6.0.4": "patches/@changesets__assemble-release-plan.patch",
>       "eslint-plugin-deprecation@3.0.0": "patches/eslint-plugin-deprecation.patch",
>       "tsup@8.3.5": "patches/tsup@8.3.5.patch"
>     }
>   }
> ```

✅ fixed

> Screen.Recording.2024-12-20.at.00.41.51.mov 
> need to fix images statically imported in cards
> 
> https://github.com/shuding/nextra/blob/ac70dfedfec29f3634a333a282ee94c1e0765cb5/docs/pages/showcase.mdx?plain=1#L12-L14

✅ fixed

**Comment by dimaMachina** [comment]
white flash on the dark theme should be fixed as well 

https://github.com/user-attachments/assets/461a9d1f-f647-4361-8587-611eaf723df3

**Comment by dimaMachina** [comment]
> white flash on the dark theme should be fixed as well
> 
>  Screen.Recording.2024-12-20.at.01.16.38.mov

✅ fixed as well

**Review by dimaMachina — APPROVED** [review]
Ok, I took the initiative and fixed all found issues, this feature is ready to ship in the minor v3 version

Please add a changeset with `minor` version and update docs to describe this new behaviour in the https://nextra.site/docs/guide/image page
And also describe how to disable zoom globally


```js
import { Image } from 'nextra/components'

// ... override mdx components 
components: {
  img: Image
}
```

Thanks for contributing!

**Comment by jellli** [comment]
> Ok, I took the initiative and fixed all found issues, this feature is ready to ship in the minor v3 version
> 
> Please add a changeset with `minor` version and update docs to describe this new behaviour in the https://nextra.site/docs/guide/image page And also describe how to disable zoom globally
> 
> ```js
> import { Image } from 'nextra/components'
> 
> // ... override mdx components 
> components: {
>   img: Image
> }
> ```
> 
> Thanks for contributing!

Whoa, thx for the review! 
Yes, I will add docs about these changes.

File: docs/pages/docs/guide/image.mdx [hunk 6]
```diff
@@ -39,3 +39,46 @@ With Next.js Image, there will be no layout shift, and a beautiful blurry
...
+### Disable Image Zoom
+
+For `nextra-docs-theme` and `nextra-blog-theme`, you can disable image zoom by
+replacing the `img` component used in MDX.
+
+```jsx filename="theme.config.jsx"
+import { Image } from 'nextra/components'
+
+export default {
+  // ...your other configurations
```
Comment (dimaMachina) [inline docs/pages/docs/guide/image.mdx:59]:
```suggestion
  // ... your other configurations
```

**Review by dimaMachina — APPROVED** [review]
thank you!

**Pull request merged**